### ISSUE TITLE
AzureRMWebAppDeployment XML Transformations

### ISSUE THREAD

**Issue by seanrockster** [Microsoft/vsts-tasks#3507]
Glad to see this in v3 however it doesn't work as one would expect. Or maybe not working how i'm used to given we use octopus deploy currently.

VS will include the environment *.config files as _DependentUpon_ the web.config in the project and that is fine. However despite the additional *.config files having their Build Action property set to Content, aside from Web.Release.config, none of the others are included in the artifact (in my case the published web application). Correct me if i'm wrong but i'm assuming this is due to the build action and MSBuild will do the transformation at that time and throw away any other *.configs?

Anyway, i've gotten around this by removing the dependency, that is opening the csproj file and removing the DependentUpon child element of the additional *.config files. Of course the *.config files are deloyed with the site but that is easily fixed with a ps script.

Am i missing something?

**Comment by bryanmacfarlane** [comment]
It's unclear which task in this repo you're creating this issue for.

Which one of these are you having an issue with: https://github.com/Microsoft/vsts-tasks/tree/master/Tasks

**Comment by seanrockster** [comment]
This one
https://github.com/Microsoft/vsts-tasks/tree/master/Tasks/AzureRmWebAppDeployment

And under releaseNotes here https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/AzureRmWebAppDeployment/task.loc.json

It states whats new in 3.0, e.g. Supports File Transformations (XDT) etc

To be fair, it does work, provided those *.config files are present. But as i said above, msbuild appears to exclude those from the artifact so maybe this is really relevant to that project and not this.

**Comment by bryanmacfarlane** [comment]
routing to release team for their call on it.

**Comment by seanrockster** [comment]
Another issue I'm having is the xml transformations do appear to work for the _connectionStrings_ element. In the same *.config file I can see it working (provided the *.config is in the artifact) for the _appSettings_ element but not the _connectionStrings_ element. I've tried with and without a vs build config for the target environment, in my case i called it Test.

**Comment by arjgupta** [comment]
Hi @seanrockster , is the connection string issue that you are facing similar to this: #3495

**Comment by seanrockster** [comment]
@arjgupta My issue is xml transformations, i.e. xml files included with the app, rather than substitutions (key/values specified in azure). However it seems using the app settings feature in azure is the preferred method - and in fact does work for us. However it only handles appSettings and connectionStrings.

**Comment by arjgupta** [comment]
Hi @seanrockster , please send the full debug logs to RM_Customer_Queries@microsoft.com. Also, what kind of xml transformation are you trying to apply and what is the syntax that you are using for it?

**Comment by gorillapower** [comment]
@seanrockster did this get resolved? Im experiencing similar issues where Ive included my *.config transforms in an artifact, published the artifact and then  applied xml transformations in a Release environment definition. To no avail.

**Comment by seanrockster** [comment]
Not as far as I'm aware.

**Comment by niadak** [comment]
@seanrockster, @gorillapower, Hi, if you are still facing any issue please send the logs to  RM_Customer_Queries@microsoft.com.

@gorillapower , The XML transformations are applied on all the *.config files which are present inside the zip package. Please make sure your *.config file is inside the zip.

**Comment by Ajay-MS** [comment]
@gorillapower  
As per internal thread I assume you are unblocked on issue. 

@seanrockster 
If you are still unblocked on issue, please send the full debug logs to RM_Customer_Queries@microsoft.com. Also, what kind of xml transformation are you trying to apply and what is the syntax that you are using for it?
You can generate the debug logs by setting the system.debug variable to true in the variables tab for the release definition

**Comment by gorillapower** [comment]
Hi Ajay,

Yes thanks. It is important to note that the transform file must be in the
same directory as your web.config file for this to work. That was my issue.

**Comment by Ajay-MS** [comment]
@seanrockster 

As I didn't find any logs from your end I am hoping you are unblocked on issue. If you still face this feel free to reopen issue again.
